all/bodies/vision.py

```diff
@@ -9,23 +9,55 @@ def __init__(self, agent, size=4, lazy=False):
         self._frames = []
         self._size = size
         self._lazy = lazy
+        self._to_cache = ToCache()
 
     def process_state(self, state):
         if not self._frames:
             self._frames = [state.observation] * self._size
         else:
             self._frames = self._frames[1:] + [state.observation]
         if self._lazy:
-            return LazyState.from_state(state, self._frames)
+            return LazyState.from_state(state, self._frames, self._to_cache)
         if isinstance(state, StateArray):
             return state.update('observation', torch.cat(self._frames, dim=1))
         return state.update('observation', torch.cat(self._frames, dim=0))
 
 
+class ToCache:
```

Comment:
Perhaps it would be good to add a docstring explaining the purpose of this? It probably would not be immediately obvious to somebody reading the code for the first time.

Reply:
Added a docstring